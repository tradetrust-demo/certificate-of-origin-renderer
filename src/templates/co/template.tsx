import React from "react";
import { get } from "lodash";
import { TemplateProps } from "@govtechsg/decentralized-renderer-react-components";

const borderStyle = {
  borderStyle: "solid",
  borderWidth: 0.5,
  borderColor: "#FFF"
};

const ExporterAndProducer = (document: any) => (
  <div className="row">
    <div className="col-12 col-md-6">
      <div className="p-2" style={borderStyle}>
        <div>1. Exporter{"'"}s name, address and country:</div>
        <pre className="p-2">
          <div>{get(document, "exporter.name")}</div>
          <div>{get(document, "exporter.address.street")}</div>
          <div>{`${get(document, "exporter.address.city")}, ${get(document, "exporter.address.state")}, ${get(
            document,
            "exporter.address.country"
          )}`}</div>
        </pre>
      </div>
      <div className="p-2" style={borderStyle}>
        <div>2. Producer{"'"}s name and address (if known):</div>
        <pre className="p-2">
          <div>{get(document, "producer.name")}</div>
          <div>{get(document, "producer.address.street")}</div>
          <div>{`${get(document, "producer.address.city")}, ${get(document, "producer.address.state")}, ${get(
            document,
            "producer.address.country"
          )}`}</div>
        </pre>
      </div>
    </div>
    <div className="col-12 col-md-6" style={borderStyle}>
      <div className="d-flex flex-column justify-content-between h-100">
        <div className="p-2">Certificate No.: {document.id}</div>
        <div className="align-self-center">
          <div className="text-center p-2">
            <strong>CERTIFICATE OF ORIGIN</strong>
          </div>
          <div className="text-center p-2">
            <strong>Form for China-Australia Free Trade Agreement</strong>
          </div>
        </div>
        <div className="p-2">Issued in: {get(document, "issuers[0].address.country")}&nbsp;</div>
      </div>
    </div>
  </div>
);

const Importer = (document: any) => (
  <div className="row">
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      <div>3. Importer{"'"}s name, address and country (if known):</div>
      <pre className="p-2">
        <div>{get(document, "importer.name")}</div>
        <div>{get(document, "importer.address.street")}</div>
        <div>{`${get(document, "importer.address.city")}, ${get(document, "importer.address.state")}, ${get(
          document,
          "importer.address.country"
        )}`}</div>
      </pre>
    </div>
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      For official use only
    </div>
  </div>
);

const TransportAndPort = (document: any) => (
  <div className="row">
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      <div>4. Means of transport and route (if known):</div>
      <div className="p-2">
        <div>
          Departure date: <pre>{get(document, "transport.departureDate")}</pre>
        </div>
        <div>
          Vessel/Flight/Train/Vehicle No.: <pre>{get(document, "transport.vehicleNo")}</pre>
        </div>
        <div>
          Port of loading: <pre>{get(document, "transport.loadingPort")}</pre>
        </div>
        <div>
          Port of discharge: <pre>{get(document, "transport.dischargePort")}</pre>
        </div>
      </div>
    </div>
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      Remarks
    </div>
  </div>
);

const LineItems = (item: any, key: any) => {
  return (
    <div className="row" key={key}>
      <div className="col-1 p-2" style={borderStyle}>
        {key + 1}
      </div>
      <div className="col-2 p-2" style={borderStyle}>
        {get(item, "mark", "")}
      </div>
      <div className="col-3 p-2" style={borderStyle}>
        {get(item, "description", "")}
      </div>
      <div className="col-1 p-2" style={borderStyle}>
        {get(item, "hscode", "")}
      </div>
      <div className="col-1 p-2" style={borderStyle}>
        {get(item, "originCriterion", "")}
      </div>
      <div className="col-2 p-2" style={borderStyle}>
        {`${get(item, "quantity.value")} ${get(item, "quantity.unit")}`}
      </div>
      <div className="col-2 p-2" style={borderStyle}>
        {`${get(item, "invoiceNo")} ${get(item, "invoiceDate")}`}
      </div>
    </div>
  );
};

const ItemsTransported = (document: any) => {
  const content = document.items.map(LineItems);
  return (
    <>
      <div className="row">
        <div className="col-1 p-2" style={borderStyle}>
          6. Item number (max.20)
        </div>
        <div className="col-2 p-2" style={borderStyle}>
          7. Marks and numbers on packages (optional)
        </div>
        <div className="col-3 p-2" style={borderStyle}>
          8. Number and kind of packages; description of goods
        </div>
        <div className="col-1 p-2" style={borderStyle}>
          9. HS code (6-digit code)
        </div>
        <div className="col-1 p-2" style={borderStyle}>
          10. Origin criterion
        </div>
        <div className="col-2 p-2" style={borderStyle}>
          11. Gross or net weight or other quantity (e.g. Quantity Unit, litres, m<sup>3</sup>.)
        </div>
        <div className="col-2 p-2" style={borderStyle}>
          12. Invoice number and date
        </div>
      </div>
      {content}
    </>
  );
};

const Declarations = (document: any) => (
  <div className="row">
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      <div className="d-flex flex-column justify-content-between h-100">
        <div>13. Decaration by the exporter or producer</div>
        <div className="p-2">
          The undersigned herby declares that the above-stated information is correct and that the goods exported to{" "}
          {get(document, "importer.address.country")} comply with the origin requirements specified in the
          China-Australia Free Trade Agreement.
        </div>
        <div className="p-2">
          <div>
            {`${get(document, "issuers[0].address.city")} ${get(document, "issuers[0].address.country")} ${get(
              document,
              "issuedDate"
            )}`}
          </div>
          <div>Place, date and signature of authorised person</div>
        </div>
      </div>
    </div>
    <div className="col-12 col-md-6 p-2" style={borderStyle}>
      <div className="p-2">14. Certification</div>
      <div className="p-2">
        On the basis of the control carried out, it is hereby certified that the information herein is correct and that
        the described goods comply with the origin requirements of the China-Australia Free Trade Agreement.
      </div>
      <div className="p-2">
        <div>
          {`${get(document, "issuers[0].address.city")} ${get(document, "issuers[0].address.country")} ${get(
            document,
            "issuedDate"
          )}`}
        </div>
        <div>Place, date and signature of authorised person</div>
      </div>
      <div className="p-2 row">
        <div className="col-6">Tel: {get(document, "issuers[0].contact.tel")}</div>
        <div className="col-6">Fax: {get(document, "issuers[0].contact.fax")}</div>
      </div>
      <div className="p-2">
        <div>Address:</div>
        <div>{get(document, "issuers[0].address.street")}</div>
        <div>{`${get(document, "issuers[0].address.city")} ${get(document, "issuers[0].address.state")} ${get(
          document,
          "issuers[0].address.country"
        )}`}</div>
      </div>
    </div>
  </div>
);

export const Template: React.FunctionComponent<TemplateProps<any>> = ({ document }) => (
  <div className="container p-2" style={{ borderStyle: "solid", borderWidth: 1 }}>
    <div className="text-center">CERTIFICATE OF ORIGIN</div>
    <div className="p-2" style={{ backgroundColor: "#e9f0c4", ...borderStyle }}>
      {ExporterAndProducer(document)}
      {Importer(document)}
      {TransportAndPort(document)}
      {ItemsTransported(document)}
      {Declarations(document)}
    </div>
  </div>
);
