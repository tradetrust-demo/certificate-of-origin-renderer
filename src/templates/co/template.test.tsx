import React from "react";
import renderer from "react-test-renderer";
import { Template } from "./template";

describe("default template", () => {
  it("matches snapshot", () => {
    const document = {
      id: "A182470ZC43306037",
      $template: {
        name: "CERTIFICATE_OF_ORIGIN",
        type: "EMBEDDED_RENDERER",
        url: "https://demo-co.openattestation.com"
      },
      issuedDate: "2018-01-09",
      issuers: [
        {
          name: "DEMO STORE",
          address: {
            street: "135 DONNYBROOK RD",
            city: "MELBOURNE",
            state: "VICTORIA",
            country: "AUSTRALIA"
          },
          contact: {
            tel: "+61-3-93911627",
            fax: "+61-3-93911687"
          },
          documentStore: "0x9178F546D3FF57D7A6352bD61B80cCCD46199C2d"
        }
      ],
      exporter: {
        name: "FINE RED'S",
        address: {
          street: "90 COLLINS ST",
          city: "MELBOURNE",
          state: "VICTORIA",
          country: "AUSTRALIA"
        }
      },
      producer: {
        name: "PERFECT PINOT LTD",
        address: {
          street: "309 KING ST",
          city: "MELBOURNE",
          state: "VICTORIA",
          country: "AUSTRALIA"
        }
      },
      importer: {
        name: "HUNAN WINES",
        address: {
          street: "EAST JIEFANG ROAD",
          city: "CHANGSHA",
          state: "HUNAN",
          country: "CHINA"
        }
      },
      transport: {
        departureDate: "2018-01-11",
        vehicleNo: "CZ3073/11-JAN",
        loadingPort: "MELBOURNE, AUSTRALIA",
        dischargePort: "HUNAN, CHINA"
      },
      remarks: "",
      items: [
        {
          marks: "",
          description: "16667 CARTONS OF RED WINE",
          hsCode: "220429",
          originCriterion: "WP",
          quantity: {
            value: 5000,
            unit: "LITRES"
          },
          invoiceNo: 24082011,
          invoiceDate: "2018-01-07"
        }
      ]
    };
    const component = renderer.create(<Template document={document} handleObfuscation={() => void 0} />);
    const tree = component.toJSON();
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(tree).toMatchSnapshot();
  });
});
