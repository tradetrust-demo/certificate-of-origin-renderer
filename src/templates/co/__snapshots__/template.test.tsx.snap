// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default template matches snapshot 1`] = `
<div>
  <div
    className="container p-2"
    style={
      Object {
        "borderStyle": "solid",
        "borderWidth": 1,
      }
    }
  >
    <div
      className="text-center"
    >
      CERTIFICATE OF ORIGIN
    </div>
    <div
      className="p-2"
      style={
        Object {
          "backgroundColor": "#e9f0c4",
          "borderColor": "#FFF",
          "borderStyle": "solid",
          "borderWidth": 0.5,
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-6"
        >
          <div
            className="p-2"
            style={
              Object {
                "borderColor": "#FFF",
                "borderStyle": "solid",
                "borderWidth": 0.5,
              }
            }
          >
            <div>
              1. Exporter
              '
              s name, address and country:
            </div>
            <pre
              className="p-2"
            >
              <div>
                FINE RED'S
              </div>
              <div>
                90 COLLINS ST
              </div>
              <div>
                MELBOURNE, VICTORIA, AUSTRALIA
              </div>
            </pre>
          </div>
          <div
            className="p-2"
            style={
              Object {
                "borderColor": "#FFF",
                "borderStyle": "solid",
                "borderWidth": 0.5,
              }
            }
          >
            <div>
              2. Producer
              '
              s name and address (if known):
            </div>
            <pre
              className="p-2"
            >
              <div>
                PERFECT PINOT LTD
              </div>
              <div>
                309 KING ST
              </div>
              <div>
                MELBOURNE, VICTORIA, AUSTRALIA
              </div>
            </pre>
          </div>
        </div>
        <div
          className="col-12 col-md-6"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          <div
            className="d-flex flex-column justify-content-between h-100"
          >
            <div
              className="p-2"
            >
              Certificate No.: 
              A182470ZC43306037
            </div>
            <div
              className="align-self-center"
            >
              <div
                className="text-center p-2"
              >
                <strong>
                  CERTIFICATE OF ORIGIN
                </strong>
              </div>
              <div
                className="text-center p-2"
              >
                <strong>
                  Form for China-Australia Free Trade Agreement
                </strong>
              </div>
            </div>
            <div
              className="p-2"
            >
              Issued in: 
              AUSTRALIA
              Â 
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          <div>
            3. Importer
            '
            s name, address and country (if known):
          </div>
          <pre
            className="p-2"
          >
            <div>
              HUNAN WINES
            </div>
            <div>
              EAST JIEFANG ROAD
            </div>
            <div>
              CHANGSHA, HUNAN, CHINA
            </div>
          </pre>
        </div>
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          For official use only
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          <div>
            4. Means of transport and route (if known):
          </div>
          <div
            className="p-2"
          >
            <div>
              Departure date: 
              <pre>
                2018-01-11
              </pre>
            </div>
            <div>
              Vessel/Flight/Train/Vehicle No.: 
              <pre>
                CZ3073/11-JAN
              </pre>
            </div>
            <div>
              Port of loading: 
              <pre>
                MELBOURNE, AUSTRALIA
              </pre>
            </div>
            <div>
              Port of discharge: 
              <pre>
                HUNAN, CHINA
              </pre>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          Remarks
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          6. Item number (max.20)
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          7. Marks and numbers on packages (optional)
        </div>
        <div
          className="col-3 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          8. Number and kind of packages; description of goods
        </div>
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          9. HS code (6-digit code)
        </div>
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          10. Origin criterion
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          11. Gross or net weight or other quantity (e.g. Quantity Unit, litres, m
          <sup>
            3
          </sup>
          .)
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          12. Invoice number and date
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          1
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          
        </div>
        <div
          className="col-3 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          16667 CARTONS OF RED WINE
        </div>
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          
        </div>
        <div
          className="col-1 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          WP
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          5000 LITRES
        </div>
        <div
          className="col-2 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          24082011 2018-01-07
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          <div
            className="d-flex flex-column justify-content-between h-100"
          >
            <div>
              13. Decaration by the exporter or producer
            </div>
            <div
              className="p-2"
            >
              The undersigned herby declares that the above-stated information is correct and that the goods exported to
               
              CHINA
               comply with the origin requirements specified in the China-Australia Free Trade Agreement.
            </div>
            <div
              className="p-2"
            >
              <div>
                MELBOURNE AUSTRALIA 2018-01-09
              </div>
              <div>
                Place, date and signature of authorised person
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-6 p-2"
          style={
            Object {
              "borderColor": "#FFF",
              "borderStyle": "solid",
              "borderWidth": 0.5,
            }
          }
        >
          <div
            className="p-2"
          >
            14. Certification
          </div>
          <div
            className="p-2"
          >
            On the basis of the control carried out, it is hereby certified that the information herein is correct and that the described goods comply with the origin requirements of the China-Australia Free Trade Agreement.
          </div>
          <div
            className="p-2"
          >
            <div>
              MELBOURNE AUSTRALIA 2018-01-09
            </div>
            <div>
              Place, date and signature of authorised person
            </div>
          </div>
          <div
            className="p-2 row"
          >
            <div
              className="col-6"
            >
              Tel: 
              +61-3-93911627
            </div>
            <div
              className="col-6"
            >
              Fax: 
              +61-3-93911687
            </div>
          </div>
          <div
            className="p-2"
          >
            <div>
              Address:
            </div>
            <div>
              135 DONNYBROOK RD
            </div>
            <div>
              MELBOURNE VICTORIA AUSTRALIA
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
